python/GafferUI/PathListingWidget.py

```diff
@@ -1028,6 +1081,108 @@ def __sectionResized( self, index, oldWidth, newWidth ) :
 		column = self.__getColumn( index )
 		self.__columnWidthAdjustments[column] = newWidth - self.__idealColumnWidths[column]
 
+		# When any column is user resized, automatically adjust the last column's width
+		# in order to make use of any empty space.
+		if index == self.header().count() - 1 :
+			# We prevent the last column from being user resized narrower than the
+			# available empty space in the header, but still allow it to be enlarged
+			# by providing the user specified column width as the minimum.
+			self.__resizeLastColumnToAvailableWidth( newWidth )
+		else :
+			self.__resizeLastColumnToAvailableWidth()
+
+	def __resizeLastColumnToAvailableWidth( self, minimumSuggestedWidth = 0 ) :
+
+		header = self.header()
+		if header.stretchLastSection() :
+			# Qt's behaviour will override and extend the last section for us automatically
+			return
+
+		availableWidth = header.width()
+		lastColumn = header.count() - 1
+		for i in range( 0, lastColumn ) :
+			availableWidth -= header.sectionSize( i )
+
+		minimumSectionWidth = max( header.minimumSectionSize(), header.sectionSizeHint( lastColumn ) )
+		minimumWidth = max( minimumSectionWidth, minimumSuggestedWidth )
+
+		# take column width adjustment into account when resizing, so we don't
+		# inadvertently truncate the last column after a user has extended it
+		column = self.__getColumn( lastColumn )
+		if self.__columnWidthAdjustments[ column ] != 0 :
+			preferredWidth = self.__idealColumnWidths[ column ] + self.__columnWidthAdjustments[ column ]
+			minimumWidth = max( preferredWidth, minimumWidth )
+
+		adjustedWidth = max( minimumWidth, availableWidth )
+		self.__lastColumnWidth = adjustedWidth
+
+		self.__recalculatingColumnWidths = True
+		header.resizeSection( lastColumn, adjustedWidth )
+		self.__recalculatingColumnWidths = False
+
+	def __geometriesChanged( self ) :
+
+		self.updateGeometry()
+
+		viewportWidth = self.viewport().width()
+		if viewportWidth == self.__previousViewportWidth :
+			# We're only interested in changes to width
+			return
+
+		if self.header().length() <= self.__previousViewportWidth or self.header().length() < viewportWidth :
+			# We use the previous viewport width to help determine whether the columns should be resized,
+			# if the columns fit within the previous width, then we should maintain that relationship and
+			# resize to fit the new width. If the columns are narrower than the current viewport width,
+			# then we resize to make use of the newly available space.
+			self.__updatingGeometry = True
+			self.__resizeStretchColumns()
+			self.__updatingGeometry = False
+
+		self.__previousViewportWidth = viewportWidth
+
+	def __resizeStretchColumns( self ) :
+
+		header = self.header()
+		if header.stretchLastSection() :
+			# Qt's behaviour will override and result in the last section taking
+			# all available space so there's no need to calculate our own widths.
+			return
```

Comment:
Another one of those ones that I think should always be `False` - maybe I missed something?

Reply:
Removed in 63a7d55.